### PR TITLE
Bug 2161903: Error message on UI during failover

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Signed-off-by: Timothy Asir <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
test2-application is not assigned to any drpolicy
Screenshot from 2023-01-20 18-04-13

Screenshot from 2023-01-20 18-02-29

Peer is not ready failover
Screenshot from 2023-01-20 18-08-46

MANAGED_CLUSTERS_ARE_DOWN
Screenshot from 2023-01-20 16-16-01

TARGET_CLUSTER_IS_NOT_AVAILABLE
Screenshot from 2023-01-20 16-16-47

SOME_CLUSTERS_ARE_FENCED
Screenshot from 2023-01-20 18-19-31

PRIMARY_CLUSTER_IS_NOT_FENCED
Screenshot from 2023-01-20 17-23-43

TARGET_CLUSTER_IS_FENCED
Screenshot from 2023-01-20 17-40-08

NO_SUBSCRIPTION_GROUP_FOUND
Screenshot from 2023-01-20 17-51-59

Screenshot from 2023-01-20 16-55-27

**Comment by TimothyAsirJeyasing** [comment]
@GowthamShanmugam , @SanjalKatiyar  Please review

File: packages/mco/components/disaster-recovery/app-failover-relocate/subscription-group-selector.tsx [hunk 1]
```diff
@@ -31,7 +32,7 @@ const validateTargetCluster = (
   drPolicyControlState: DRPolicyControlState,
   targetCluster: PlacementDecision
 ) => {
-  const currentDeploymentCluster = drPolicyControlState?.clusterName;
+  const currentDeploymentCluster = "spoke" || drPolicyControlState?.clusterName;
```
Comment (GowthamShanmugam) [inline packages/mco/components/disaster-recovery/app-failover-relocate/subscription-group-selector.tsx:35]:
```suggestion
  const currentDeploymentCluster = drPolicyControlState?.clusterName;
```

Comment (TimothyAsirJeyasing) [reply packages/mco/components/disaster-recovery/app-failover-relocate/subscription-group-selector.tsx]: Done

File: packages/mco/components/disaster-recovery/app-failover-relocate/peer-cluster-status.tsx [hunk 2]
```diff
@@ -136,12 +138,11 @@ export const PeerClusterStatus: React.FC<PeerClusterStatusProps> = ({
...
-            t(
-              '{{actionType}} cannot be initiated becuase peer is not in ready state.',
-              { actionType: state.actionType }
-            )
+            state.actionType === ACTION_TYPE.FAILOVER
+              ? ErrorMessageType.PEER_IS_NOT_READY_FAILOVER
+              : ErrorMessageType.PEER_IS_NOT_READY_RELOCATE
           )
-        : setErrorMessage('');
+        : setErrorMessage(null);
```
[the remark this answers is not in the document]
Comment (TimothyAsirJeyasing) [reply packages/mco/components/disaster-recovery/app-failover-relocate/peer-cluster-status.tsx]: Done

File: packages/mco/components/disaster-recovery/app-failover-relocate/reducer.ts [hunk 3]
```diff
@@ -27,12 +28,11 @@ export type TargetClusterType = Partial<{
...
-  managedClustersErrorMessage: string;
-  targetClusterErrorMessage: string;
-  subscriptionGroupErrorMessage: string;
-  peerStatusErrorMessage: string;
-  failoverAndRelocateActionErrorMessage: string;
+  drPolicyControlStateErrorMessage: ErrorMessageType;
+  managedClustersErrorMessage: ErrorMessageType;
+  targetClusterErrorMessage: ErrorMessageType;
+  subscriptionGroupErrorMessage: ErrorMessageType;
+  peerStatusErrorMessage: ErrorMessageType;
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply packages/mco/components/disaster-recovery/app-failover-relocate/reducer.ts:35]: This change is not yet done

[the remark this answers is not in the document]
Comment (TimothyAsirJeyasing) [reply packages/mco/components/disaster-recovery/app-failover-relocate/reducer.ts]: Sorry, i have done the changes now

File: packages/mco/components/disaster-recovery/app-failover-relocate/error-messages.tsx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      <Trans t={t}>
+        <ul>
+          <li>
+            The status for your target cluster must be unfenced for initiating
+            failover. Check the status and try again.
+          </li>
+          <li>
+            To unfence your cluster, follow the instructions in the
+            documentation
+            <ViewDocumentation doclink={DOC_LINKS.UNFENCING} />.
```
Comment (GowthamShanmugam) [inline packages/mco/components/disaster-recovery/app-failover-relocate/error-messages.tsx:191]:
```suggestion
            <ViewDocumentation doclink={DOC_LINKS.FENCING} />.
```
This is due to latest document change

Comment (TimothyAsirJeyasing) [reply packages/mco/components/disaster-recovery/app-failover-relocate/error-messages.tsx]: Done

File: packages/mco/utils/doc-utils.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export const ODF_DOC_HOME = `https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/${ODF_DOC_VERSION}`;
+export const ODF_DOC_BASE_PATH = `${ODF_DOC_HOME}/html-single`;
+
+export const ACM_DOC_VERSION = '2.7';
+export const ACM_DOC_HOME = `https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/${ACM_DOC_VERSION}`;
+export const ACM_DOC_BASE_PATH = `${ACM_DOC_HOME}/html-single`;
+
+export const DOC_LINKS = {
+  APPLY_POLICY: `${ODF_DOC_BASE_PATH}/configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/index#apply-drpolicy-to-sample-application_manage-dr`,
+  FENCING: `${ODF_DOC_BASE_PATH}/configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/index#enable-fencing_manage-dr`,
```
Comment (GowthamShanmugam) [inline packages/mco/utils/doc-utils.tsx:16]:
```suggestion
  MDR_FAILOVER: `${ODF_DOC_BASE_PATH}/configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/index#application-failover-between-managed-clusters_manage-dr`,
```

File: packages/mco/utils/doc-utils.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  UNFENCING: `${ODF_DOC_BASE_PATH}/configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/index#disable-fencing_manage-dr`,
```
Comment (GowthamShanmugam) [inline packages/mco/utils/doc-utils.tsx:17]:
```suggestion
  MDR_RELOCATE: `${ODF_DOC_BASE_PATH}/configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/index#relocating-application-between-managed-clusters_manage-dr`,
```

Comment (TimothyAsirJeyasing) [reply packages/mco/utils/doc-utils.tsx]: done

File: packages/mco/utils/doc-utils.tsx (hunk 5 above)
Comment (TimothyAsirJeyasing) [reply packages/mco/utils/doc-utils.tsx]: done

**Review by GowthamShanmugam — APPROVED** [review]

**Comment by TimothyAsirJeyasing** [comment]
@GowthamShanmugam , @SanjalKatiyar Please review

**Comment by SanjalKatiyar** [comment]
/retitle [release-4.12] Bug 2161903: Error message on UI during failover

**Pull request merged**